pyrefly/lib/alt/function.rs

```diff
@@ -234,6 +256,16 @@ impl<'a, Ans: LookupAnswer> AnswersSolver<'a, Ans> {
             self_type = self_type.map(Type::type_form);
         }
 
+        if is_overload && stub_or_impl == FunctionStubOrImpl::Impl {
+            self.error(
+                errors,
+                def.name.range,
+                ErrorKind::InvalidOverload,
+                None,
+                "@overload decorator should not be used on function implementations.".to_owned(),
```

Comment:
Similar to my other comment - this is a good error message, but we should report either this error or "Overloaded function must have an implementation," not both.

If you add a new `stub_or_impl` field to DecoratedFunction (https://github.com/facebook/pyrefly/blob/4a08127ee842d006904558dca4baecf9a3b12251/pyrefly/lib/alt/types/decorated_function.rs#L29), then you can move this check to solve_function_binding and do something like:
```
if !skip_implementation {
    if def.stub_or_impl == FunctionStubOrImpl::Impl {
        // report "@overload decorator should not be used on function implementations."
    else {
        // report "Overloaded function must have an implementation"
    }
}
```